### PR TITLE
#5256 Connecting data to GA4

### PR DESCRIPTION
**Pull request by shondriane**
### Ticket(s)

- _[https://airtable.com/appfJZShN8K4tcWHU/pagxblGyhI5EJIA3v?HjEAY=recm1lmejkNoA4Pto](ticket)_

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Description

Add google tag to every page of the website. 

### Screenshots

Include relevant screenshots.

#### Before
image

#### After
image

### Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand.

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -3,21 +3,41 @@ import '../styles/global.css';
...
+import React, { useEffect } from 'react';
+import TagManager from 'react-gtm-module';
 
 import Footer from '../components/layout/Footer.tsx';
 import Header from '../components/layout/Header';
 import theme from '../styles/themes/theme.tsx';
 import createEmotionCache from '../utils/createEmotionCache';
 
 const clientSideEmotionCache = createEmotionCache();
+const gtmId = process.env.NEXT_PUBLIC_GTM_ID;
```
Comment (e-papanicolas) [inline pages/_app.js:17]: Can you add this new variable with an example value in `.env.example` please! And also update everyone in #cop-dev slack with how to get a real version for their local `.env`.

File: pages/_app.js [hunk 2]
```diff
@@ -3,21 +3,41 @@ import '../styles/global.css';
...
+
+  useEffect(() => {
+    // initializing google tag
+    TagManager.initialize({ gtmId });
+
+    // tracking every page via event listener when a route change is completed
+    const handleRouteChangeComplete = () => {
+      TagManager.initialize({ gtmId });
+    };
+    router.events.on('routeChangeComplete', handleRouteChangeComplete);
```
Comment (e-papanicolas) [inline pages/_app.js:35]: Will you please add a line of space above and below this line for better readability?

**Review by e-papanicolas — CHANGES_REQUESTED** [review]

File: pages/_app.js (hunk 1 above)
Comment (shondriane) [reply pages/_app.js:17]: Hello Eleni,
I will add the new value in .env.example. I would also send everyone in #cop-dev with the gtmId so they can update their local env file appropriately.

File: pages/_app.js (hunk 2 above)
Comment (shondriane) [reply pages/_app.js:35]: Of course, will do!

**Review by e-papanicolas — APPROVED** [review]

**Pull request merged**